tests/test_result.py

```diff
@@ -171,6 +173,30 @@ def test_map_err() -> None:
     assert n.map_err(str.upper).err() == 'NAY'
 
 
+def test_and_then() -> None:
```

Comment:
For reference, I took these test cases from the rust docs,
https://doc.rust-lang.org/std/result/enum.Result.html#examples-26